api.go

```diff
@@ -0,0 +1,1997 @@
+	return out, req.Send()
+}
+
+// ListBucketQueryWithContext is the same as ListBucketQuery with the addition of
+// the ability to pass a context and additional request options.
+func (c *S3) ListBucketQueryWithContext(ctx aws.Context, input *ListBucketQueryInput, opts ...request.Option) (*ListBucketQueryOutput, error) {
+	req, out := c.ListBucketQueryRequest(input)
+	req.SetContext(ctx)
+	req.ApplyOptions(opts...)
+	return out, req.Send()
+}
+
+const opPutBucketIsStaleAllowed = "PutBucketIsStaleAllowed"
+
+// PutBucketIsStaleAllowedRequest generates request.Request
+func (c *S3) PutBucketIsStaleAllowedRequest(input *PutBucketIsStaleAllowedInput) (req *request.Request, output *PutBucketIsStaleAllowedOutput) {
+	op := &request.Operation{
+		Name:       opPutBucketIsStaleAllowed,
+		HTTPMethod: "PUT",
+		HTTPPath:   "/{Bucket}?isstaleallowed",
+	}
+
+	if input == nil {
+		input = &PutBucketIsStaleAllowedInput{}
+	}
+
+	output = &PutBucketIsStaleAllowedOutput{}
+	req = c.newRequest(op, input, output)
+	req.Handlers.Unmarshal.Remove(restxml.UnmarshalHandler)
+	req.Handlers.Unmarshal.PushBackNamed(protocol.UnmarshalDiscardBodyHandler)
+	return
+}
+
+// PutBucketIsStaleAllowed API operation for ECS Extension
+func (c *S3) PutBucketIsStaleAllowed(input *PutBucketIsStaleAllowedInput) (*PutBucketIsStaleAllowedOutput, error) {
+	req, out := c.PutBucketIsStaleAllowedRequest(input)
+	return out, req.Send()
+}
+
+// PutBucketIsStaleAllowedWithContext is the same as PutBucketIsStaleAllowed with the addition of
+// the ability to pass a context and additional request options.
+func (c *S3) PutBucketIsStaleAllowedWithContext(ctx aws.Context, input *PutBucketIsStaleAllowedInput, opts ...request.Option) (*PutBucketIsStaleAllowedOutput, error) {
+	req, out := c.PutBucketIsStaleAllowedRequest(input)
+	req.SetContext(ctx)
+	req.ApplyOptions(opts...)
+	return out, req.Send()
+}
+
+const opPutObject = "PutObject"
+
+// PutObjectExtRequest generates a request.Request
+func (c *S3) PutObjectExtRequest(input *PutObjectInput) (req *request.Request, output *PutObjectOutput) {
+	op := &request.Operation{
+		Name:       opPutObject,
+		HTTPMethod: "PUT",
+		HTTPPath:   "/{Bucket}/{Key+}",
+	}
+
+	if input == nil {
+		input = &PutObjectInput{}
+	}
+
+	output = &PutObjectOutput{}
+	req = c.newRequest(op, input, output)
+	return
+}
+
+// PutObjectExt API operation for ECS Extension.
+func (c *S3) PutObjectExt(input *PutObjectInput) (*PutObjectOutput, error) {
+	req, out := c.PutObjectExtRequest(input)
+	return out, req.Send()
+}
+
+// PutObjectExtWithContext is the same as PutObject with the addition of
+// the ability to pass a context and additional request options.
+func (c *S3) PutObjectExtWithContext(ctx aws.Context, input *PutObjectInput, opts ...request.Option) (*PutObjectOutput, error) {
+	req, out := c.PutObjectExtRequest(input)
+	req.SetContext(ctx)
+	req.ApplyOptions(opts...)
+	return out, req.Send()
+}
+
+type CreateBucketInput struct {
```

Comment:
I looked into further, actually,  xxxInput and xxxOutput structures are located after xxxRequest functions. I'd prefer to keep it where it is. (the location sequence in api.go is all request functions --> input/output structures --> misc util functions)